File: numberGuess.java
import java.lang.Math;
import java.util.Scanner;

public class numberGuess {

    public static void main(String[] args) {
        int randomInt = (int) (Math.random() * 100);
        int attempts = 0;
        int lowerBound = 0;
        int upperBound = 100;
        Scanner inputScanner = new Scanner(System.in);
        System.out.println(randomInt);
        System.out.println("Enter a value between 0 and 100");

        while (true) {
            if (inputScanner.hasNextInt()) {
                int guess = inputScanner.nextInt();
                if (guess <= lowerBound || guess >= upperBound) {
                    System.out.println("Your guess is not within the bounds. Try again!");
                    continue;
                }
                attempts++;
                if (guess < randomInt) { /* Too low */
                    lowerBound = guess;
                    System.out.printf("Too low! Enter a value between %d and %d%n", lowerBound, upperBound);
                } else if (guess > randomInt) { /* Too high */
                    upperBound = guess;
                    System.out.printf("Too high! Enter a value between %d and %d%n", lowerBound, upperBound);

                } else { /* Right value */
                    if (attempts == 1) {
                        System.out.println("Amazing! You got it right in one attempt!");
                    } else {
                        System.out.printf("You got it right! Great work! You did it in %d attempts%n", attempts);
                    }
                    inputScanner.close();
                    return;
                }
            }
        }
    }
}

File: numberGuess.py
def main():
    from random import random
    # generate some random numbers
    randomInt = int(random()*100)
    attempts = 0
    lowerBound = 1
    upperBound = 100
    guess = input ("Guess a number between 1 and 100\n") 
    while 1==1:
        try:
            guess = int(guess)
            if guess < lowerBound or guess > upperBound:
                guess = input("That guess is outside the bounds. Guess a number between " + str(lowerBound) + " and " + str(upperBound) + "\n")
                continue
            attempts += 1
            if guess < randomInt:   #Guess too low
                lowerBound = guess
                guess = input("Too low! Guess a number between " + str(lowerBound) + " and " + str(upperBound) + "\n")
            elif guess > randomInt: #Guess too low
                upperBound = guess
                guess = input("Too high! Guess a number between " + str(lowerBound) + " and " + str(upperBound) + "\n")
            else:                   #Guess is correct
                guess = input("That's the right number! Great work! It took you " + str(attempts) + " attempt(s) to guess it.\n")
                break

        except ValueError:
            guess = input("Please guess a number between " + str(lowerBound) + " and " + str(upperBound) + "\n")

if __name__ == '__main__':
    main()